src/Tuura/Concept/Circuit/Derived.hs

```diff
@@ -199,3 +201,14 @@ internals ints = interfaceConcept $ \s ->
                  if s `elem` ints
                  then Internal
                  else Unused
+
+combinationalGate :: Eq a => Expr a -> a -> CircuitConcept a
```

Comment:
Aha, nice idea!

Perhaps, we could call it simply `function :: Eq a => Expr a -> Transition a -> CircuitConcept a` and also define `complexGate :: Eq a => Expr a -> a -> CircuitConcept a` as a composition of set and reset functions.